gapis/api/vulkan/frame_loop.go

```diff
@@ -184,8 +194,27 @@ func (f *frameLoop) Transform(ctx context.Context, id api.CmdID, cmd api.Cmd, ou
 		return
 	}
 
-	if _, ok := cmd.(*VkQueuePresentKHR); ok {
+	// Skip the destroy calls for resources that are not created during loop
+	switch cmd.(type) {
```

Comment:
WE have to be very careful about these.
Vulkan is allow re-using handles, so vkDestroyBuffer, vkCreateBuffer may return the same handle. This means we may end up remapping them to the same value. I expect weird things to happen here. I would remove these DestroyBuffer fixes for now, or even better, *if* we destroy a buffer, and then create a new one with the same handle, throw an error. That way we can see how often that happens (if ever)